File: 429-n-ary-tree-level-order-traversal/429-n-ary-tree-level-order-traversal.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

class Solution:
    def levelOrder(self, root: 'Node') -> List[List[int]]:
        if not root : 
            return []
        
        ans, queue = [], [root]
        
        while queue :
            row = []
            N = len(queue)
            
            for _ in range(N) :
                t = queue.pop(0)

                for node in t.children :
                    queue.append(node)
                row.append(t.val)
            ans.append(row)
        
        return ans

File: 429-n-ary-tree-level-order-traversal/429-n-ary-tree-level-order-traversal.cpp
/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val) {
        val = _val;
    }

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
public:
    vector<vector<int>> levelOrder(Node* root) {
        vector<vector<int>> ans;
        if (!root)
            return ans;
        queue<Node*> q;
        q.push(root);
        
        while (!q.empty()) {
            vector<int> row;
            int N = q.size();
            for (int i = 0; i < N; i++) {
                Node* t = q.front();
                q.pop();
                for (auto nbr : t->children)
                    q.push(nbr);
                row.push_back(t->val);
            }
            ans.push_back(row);
        }
        
        return ans;
    }
};